community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/executionplan/StandardInternalExecutionResult.scala

```diff
@@ -46,7 +46,14 @@ abstract class StandardInternalExecutionResult(context: QueryContext,
   protected val isGraphKernelResultValue = context.isGraphKernelResultValue _
   private val scalaValues = new ScalaResultValueConverter(isGraphKernelResultValue)
 
-  private var successful = false
+  // TODO: This is an undocumented and unused
+  //       mechanism for marking a result as "successful",
+  //       I don't know what the purpose of this is, but since it
+  //       was always 'false', every cypher statement using this
+  //       would force rollback on the current transaction.
+  //       Need to sort out what the intention of this mechanism
+  //       is before merging this and removing or fixing it.
+  private var successful = true
```

Comment:
I've fixed an error related to this recently. The idea (I think) is that you should call `success()` when done in the extending classes.